app/modules/parsing/graph_construction/code_graph_service.py

```
@@ -143,7 +180,7 @@ def _store_graph(
             # Step 3: Batch insert nodes
             node_insert_start = time.time()
             logger.info(
-                f"[GRAPH GENERATION] Step 4/4: Inserting {node_count} nodes into Neo4j",
+                f"[GRAPH GENERATION] Step 3/4: Inserting {node_count} nodes into Neo4j",
```

Ruff (0.16.0)

[warning] Logging statement uses f-string

(G004)